Library/Homebrew/cmd/shellenv.sh

```diff
@@ -21,7 +21,7 @@ homebrew-shellenv() {
       echo "set -gx HOMEBREW_PREFIX \"${HOMEBREW_PREFIX}\";"
       echo "set -gx HOMEBREW_CELLAR \"${HOMEBREW_CELLAR}\";"
       echo "set -gx HOMEBREW_REPOSITORY \"${HOMEBREW_REPOSITORY}\";"
-      echo "set -q PATH; or set PATH ''; set -gx PATH \"${HOMEBREW_PREFIX}/bin\" \"${HOMEBREW_PREFIX}/sbin\" \$PATH;"
+      echo "fish_add_path -gP \"${HOMEBREW_PREFIX}/bin\" \"${HOMEBREW_PREFIX}/sbin\";"
```

Comment:
Why global and not universal?

Reply:
A universal $PATH is a bad idea (and fish_add_path doesn't allow it).

It *could* use the universal $fish_user_paths, but that's typically meant for, well, the user's stuff. It's also harder to remove because you now need to not just remove the `fish_add_path` call but also the elements from the universal. (universal variables have their issues and we're rethinking them)

Plus everything else here is global.

Reply:
And e.g. if $HOMEBREW_PREFIX changes this wouldn't remove the old paths from $fish_user_paths.